mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -278,7 +278,8 @@ class DenseMatrix @Since("1.3.0") (
   }
 
   override def hashCode: Int = {
-    com.google.common.base.Objects.hashCode(numRows : Integer, numCols: Integer, toArray)
+    val state = Seq(numRows, numCols, Arrays.hashCode(values), isTransposed.hashCode)
```

Comment:
Crazy, you might ask on the breeze list if you have time, to make sure we're not missing something, but that seems like a valid test.

We can avoid using either Breeze method in these implementations in order to write a version that we know is consistent and correct. On the other hand, it's likely that `==` is correct, and it's just that hash code isn't implemented. I suppose it should be possible to write a hash code that must be consistent with the plain meaning of equality for matrices, by making it a function of all (non-zero) elements and their positions or something. That is, is your `hashCode` going to be consistent with Breeze `==`? that's the important thing to guarantee.